votingpool/db.go

```diff
@@ -56,13 +59,79 @@ type dbSeriesRow struct {
 	privKeysEncrypted [][]byte
 }
 
+// getUsedAddrBucketID returns the used addresses bucket ID for the given series
+// and branch. It has the form seriesID:branch.
+func getUsedAddrBucketID(seriesID uint32, branch Branch) []byte {
+	return bytes.Join(
+		[][]byte{uint32ToBytes(seriesID), uint32ToBytes(uint32(branch))}, []byte(":"))
+}
+
+// putUsedAddrHash adds an entry (key==index, value==encryptedHash) to the used
+// addresses bucket of the given pool, series and branch.
+func putUsedAddrHash(tx walletdb.Tx, poolID []byte, seriesID uint32, branch Branch,
+	index Index, encryptedHash []byte) error {
+
+	usedAddrs := tx.RootBucket().Bucket(poolID).Bucket(usedAddrsBucketName)
+	bucket, err := usedAddrs.CreateBucketIfNotExists(getUsedAddrBucketID(seriesID, branch))
+	if err != nil {
+		return newError(ErrDatabase, "failed to store used address hash", err)
+	}
+	return bucket.Put(uint32ToBytes(uint32(index)), encryptedHash)
+}
+
+// getUsedAddrHash returns the addr hash with the given index from the used
+// addresses bucket of the given pool, series and branch.
+func getUsedAddrHash(
+	tx walletdb.Tx, poolID []byte, seriesID uint32, branch Branch, index Index) []byte {
+
+	usedAddrs := tx.RootBucket().Bucket(poolID).Bucket(usedAddrsBucketName)
```

Comment:
I haven't reviewed far enough to see if the pool id existence is checker before calling this function, but this will cause a panic when the id doesn't exist.